main.go

```diff
@@ -2,39 +2,28 @@ package main
 
 import (
 	"bytes"
-	"context"
 	"encoding/json"
 	"fmt"
 	"os"
 
-	"github.com/cheggaaa/pb/v3"
-	ipld "github.com/ipfs/go-ipld-format"
 	madns "github.com/multiformats/go-multiaddr-dns"
 
-	// importing so we can traverse dag-cbor and dag-json nodes in the `bitswap get` command
-	_ "github.com/ipld/go-ipld-prime/codec/dagcbor"
-	_ "github.com/ipld/go-ipld-prime/codec/dagjson"
-
 	vole "github.com/ipfs-shipyard/vole/lib"
 	"github.com/urfave/cli/v2"
 
-	"github.com/ipfs/boxo/bitswap"
-	bsnet "github.com/ipfs/boxo/bitswap/network"
-	"github.com/ipfs/boxo/blockservice"
-	blockstore "github.com/ipfs/boxo/blockstore"
-	"github.com/ipfs/boxo/ipld/merkledag"
 	"github.com/ipfs/go-cid"
-	"github.com/ipfs/go-datastore"
-	"github.com/ipfs/go-datastore/sync"
-	"github.com/libp2p/go-libp2p"
-	mplex "github.com/libp2p/go-libp2p-mplex"
-	rhelp "github.com/libp2p/go-libp2p-routing-helpers"
 	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/libp2p/go-libp2p/core/protocol"
+	"github.com/libp2p/go-libp2p/p2p/protocol/identify"
 	"github.com/multiformats/go-multiaddr"
 	"github.com/multiformats/go-multibase"
 )
 
+func init() {
+	// Lets us discover our own public address with a single observation
+	identify.ActivationThresh = 1
```

Comment:
In init so we don't set this when running tests. That's a data race. It's a race because we read it from one host, while setting it from another host. Really this should be an option in go-libp2p, but that's a slightly more involved change and I'm not sure if worth it for just this one use case. If this comes up again I'll make the change though!